server/api/student/student.controller.js

```diff
@@ -84,6 +87,100 @@ exports.outreachCounts = function(req, res) {
   });
 };
 
+/**
+ * Get summary of outreaches.
+ * restriction: 'teacher'
+ *
+ * Returns an aggregation for records based on the req user role:
+ * - teachers will get outreach summary for assignment school
+ * - manager+ will get outreach summary for all schools
+ */
+exports.outreachSummary = function(req, res) {
+  var pipeline = [{
+   $group: {
+     _id: {student: '$student', type: '$type'},
+     count: {$sum: 1}
+   }
+  }, {
+   $group: {
+     _id: '$_id.student',
+     counts: {$push: {type: '$_id.type', count: '$count'}}
+   }
+  }, {
+    $project: {
+      _id: 0,
+      student: '$_id',
+      counts: 1
+    }
+  }];
+  if (req.user.role === 'teacher') {
+    pipeline.unshift({
+       $match: {school: req.user.assignment}
+    });
+  }
+  Outreach.aggregate(pipeline, function(err, results) {
+    if (err) handleError(res, err);
+    // Convert counts array to object, generate total intervention property
+    results = _.map(results, function(result) {
+      result.counts = _.keyBy(result.counts, 'type');
+      var total = _.map(result.counts, 'count');
+      result.total = total.reduce(function(prev, curr) {
+        return prev + curr;
+      });
+      return result;
+    });
```

Comment:
Tried to use the callback in `_.keyBy` to convert object to a `[<type>]: <count>` format due to the redundancy of having a `type` property inside a key / value pair with a key of the same value.